src/v/cluster/drain_manager.cc

```diff
@@ -131,19 +131,31 @@ ss::future<> drain_manager::task() {
         }
 
         const auto draining = _draining;
-        try {
-            if (draining) {
-                co_await do_drain();
-            } else {
-                co_await do_restore();
+        while (true) {
```

Comment:
can you explain why is this required ?  I see draining is already being retried in `do_drain` method implementation

Reply:
Right, I don't think this `try / catch` is necessary but it takes effort to increase certainty up to a level to remove it because if we're wrong then the uncaught exception will kill the maintenance fiber. My motivation was not to mark draining as finished when there is a error but retry until success.

Reply:
i think this is not required and we can drop this commit not to increase the code churn and complexity.

Reply:
we agreed, that we will simply drop the try/catch